### ISSUE TITLE
Amplify v2.x doesn't support passwords with spaces

### ISSUE THREAD

**Issue by valzevul** [aws-amplify/amplify-swift#3129]
### Describe the bug

We've recently upgraded `amplify-swift` from a `1.30.3` version to `2.12.0`, and seems like since the update `Amplify.Auth.signUp` throws an error when signing up with a password containing non-leading, non-trailing space characters.

Sign in works fine though, so the issue seems to affect only sign up calls.

### Steps To Reproduce

```swift
1. Given a default configuration, call `Amplify.Auth.signUp(username:email:password)` with a password that contains a space in the middle (e.g "hello amplify!").
2. The call throws an error

AuthError: Password is required to signUp
Recovery suggestion: Make sure that a valid password is passed for signUp
  ▿ validation : 4 elements
    - .0 : "password"
    - .1 : "Password is required to signUp"
    - .2 : "Make sure that a valid password is passed for signUp"
    - .3 : nil
```

### Expected behavior

[As per the docs](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-policies.html), non-leading, non-trailing space characters should be allowed in passwords.

### Amplify Framework Version

2.12.0

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5.7

### CLI version

8.2.0

### Xcode version

14.3.1 (14E300c)

### Relevant log output

```shell
n/a
```

### Is this a regression?

Yes

### Regression additional context

The same scenario works as expected on `1.30.3`.

### Platforms

iOS

### OS Version

15.4.1

### Device

iPhone Xs

### Additional context

The issue could be reproduced on latest simulators as well

**Comment by valzevul** [comment]
As far as I can see, the error comes from the logic in `SignUpPasswordValidator.swift`:
```swift
} else if password.rangeOfCharacter(from: .whitespacesAndNewlines) != nil {
  error = .invalidPassword(message: "password includes disallowed characters")
}
```
that leads to
```
▿ Optional<SignUpError>
  ▿ some : SignUpError
    ▿ invalidPassword : 1 element
      - message : "password includes disallowed characters"
```

It also seems like there is a unit test making sure that passwords with spaces in the middle are indeed incorrect, which seems to conflict with the docs / user pool settings:

https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/Support/SignUpPasswordValidatorTests.swift#L25

**Comment by harsh62** [comment]
@valzevul I opened a PR to fix this issue. Would you be able to validate if the open PR fixes your issue?

**Comment by valzevul** [comment]
@harsh62 thanks a lot, building the app against `fix/3129-fix` via SPM seems to be solving the issue!

**Comment by ruisebas** [comment]
This was released in [2.15.3](https://github.com/aws-amplify/amplify-swift/releases/tag/2.15.3)
